### PR TITLE
[Debugger Plugin] Beautify initial code snippets

### PR DESCRIPTION
**Pull request by caisq**
* The syntax highlighting is done manually.
* Add code example for Keras.

### CODE REVIEW

**Comment by caisq** [comment]
See screenshot:
beautify_code_tdp

File: tensorboard/plugins/debugger/tf_debugger_dashboard/tf-debugger-initial-dialog.html [hunk 1]
```diff
@@ -73,6 +90,13 @@ <h2 id='dialog-title'>Waiting for first Session.run() to connect...</h2>
       margin: 10px;
       font-family: monospace;
     }
+    .code-example-section {
+      padding-bottom: 15px;
+    }
+    .code-snippet {
+      position: relative;
```
Comment (chihuahua) [inline tensorboard/plugins/debugger/tf_debugger_dashboard/tf-debugger-initial-dialog.html]: Hmm, I wonder why `position:relative` is necessary, but OK!

File: tensorboard/plugins/debugger/tf_debugger_dashboard/tf-debugger-initial-dialog.html [hunk 2]
```diff
@@ -28,24 +29,36 @@
...
+
+hook <span style='color:#308080; '>=</span> tf_debug<span style='color:#308080; '>.</span>TensorBoardDebugHook<span style='color:#308080; '>(</span><span style='color:#1060b6; '>"[[_host]]:[[_port]]"</span><span style='color:#308080; '>)</span>
+my_estimator<span style='color:#308080; '>.</span>fit<span style='color:#308080; '>(</span>x<span style='color:#308080; '>=</span>x_data<span style='color:#308080; '>,</span> y<span style='color:#308080; '>=</span>y_data<span style='color:#308080; '>,</span> steps<span style='color:#308080; '>=</span><span style='color:#008c00; '>1000</span><span style='color:#308080; '>,</span> monitors<span style='color:#308080; '>=</span><span style='color:#308080; '>[</span>hook<span style='color:#308080; '>]</span><span style='color:#308080; '>)</span></pre>
+        </div>
+        <div class="code-example-section">
+          <a href="https://www.tensorflow.org/api_docs/python/tf/keras/models" target="_blank">Keras Model</a>s:
+          <pre class="code-snippet" style='color:#000020;background:#f6f8ff;'><span style='color:#200080; font-weight:bold; '>import</span> tensorflow <span style='color:#200080; font-weight:bold; '>as</span> tf
```
Comment (nfelt) [inline tensorboard/plugins/debugger/tf_debugger_dashboard/tf-debugger-initial-dialog.html]: This snippet includes `import tensorflow as tf` but the first one for tf.Session doesn't.  Probably best if they're consistent.

File: tensorboard/plugins/debugger/tf_debugger_dashboard/tf-debugger-initial-dialog.html [hunk 3]
```diff
@@ -28,24 +29,36 @@
...
+          <pre class="code-snippet" style='color:#000020;background:#f6f8ff;'><span style='color:#200080; font-weight:bold; '>from</span> tensorflow<span style='color:#308080; '>.</span>python <span style='color:#200080; font-weight:bold; '>import</span> debug <span style='color:#200080; font-weight:bold; '>as</span> tf_debug
+
+sess <span style='color:#308080; '>=</span> tf<span style='color:#308080; '>.</span>Session<span style='color:#308080; '>(</span><span style='color:#308080; '>)</span>
+sess <span style='color:#308080; '>=</span> tf_debug<span style='color:#308080; '>.</span>TensorBoardDebugWrapperSession<span style='color:#308080; '>(</span>sess<span style='color:#308080; '>,</span> <span style='color:#1060b6; '>"[[_host]]:[[_port]]"</span><span style='color:#308080; '>)</span>
+sess<span style='color:#308080; '>.</span>run<span style='color:#308080; '>(</span>my_fetches<span style='color:#308080; '>)</span></pre>
+        </div>
+        <div  class="code-example-section">
+          <a href="https://www.tensorflow.org/programmers_guide/estimators" target="_blank">Estimator</a>s
```
Comment (nfelt) [inline tensorboard/plugins/debugger/tf_debugger_dashboard/tf-debugger-initial-dialog.html]: nit: add colon ":" to end of line for consistency with other two

File: tensorboard/plugins/debugger/tf_debugger_dashboard/tf-debugger-initial-dialog.html [hunk 4]
```diff
@@ -28,24 +29,36 @@
...
-sess.run(my_fetches)
-        </pre>
-        <pre id="hook-code-example">
-# To connect to the debugger using hooks, e.g., from tf.Estimator:
-from tensorflow.python import debug as tf_debug
-hook = tf_debug.TensorBoardDebugHook("[[_host]]:[[_port]]")
-my_estimator.fit(x=x_data, y=y_data, steps=1000, monitors=[hook])
-        </pre>
+        <div class="code-example-section">
+          <a href="https://www.tensorflow.org/api_docs/python/tf/Session" target="_blank">tf.Session</a>s:
```
Comment (nfelt) [inline tensorboard/plugins/debugger/tf_debugger_dashboard/tf-debugger-initial-dialog.html]: Hmm, this used to say "To connect to the debugger from your tf.Session:" but now we just get "tf.Session:" - I think a little more instruction might be clearer?  E.g. perhaps just something like:

To connect with a tf.Session():
To connect with an Estimator and MonitoredSession:
To connect with a Keras model:

**Comment by jart** [comment]
> Hmm, the syntax highlighting is pretty but the code snippets are a bit harder to read/edit in the source now

+1 to Nick's concern.

Syntax highlighting, while nice, should ideally be done with a library, which we might not be a point where we need yet.

**Comment by nfelt** [comment]
Hm, I think the part adding the keras code example is still useful though, even if we don't want to add the syntax highlighting just yet?  Justine is OOO for a bit so we may not be able to get her thoughts, but @caisq I think if you wanted to just resend the keras code example and wording changes that should be fine to merge.

**Comment by caisq** [comment]
@nfelt OK. Let me sent a PR with only the code.

Has there been any past conversation about adding a syntax highlighting library to TB? It seems to be potentially useful in quite a few places in TB.

**Comment by nfelt** [comment]
Sounds good, you can also just reopen this PR with new commits if you want.  I'm not aware of any prior conversation about adding syntax highlighting - I agree it's potentially useful but we'd want to balance that against an extra dep (though I think we'd want to do it in javascript so it at least wouldn't be another pip dependency).

**Pull request closed without merging**